x/mongo/driver/driver.go

```diff
@@ -73,13 +74,14 @@ type Authenticator interface {
 
 // Cred is a user's credential.
 type Cred struct {
-	Source              string
-	Username            string
-	Password            string
-	PasswordSet         bool
-	Props               map[string]string
-	OIDCMachineCallback OIDCCallback
-	OIDCHumanCallback   OIDCCallback
+	Source                 string
+	Username               string
+	Password               string
+	PasswordSet            bool
+	Props                  map[string]string
+	OIDCMachineCallback    OIDCCallback
+	OIDCHumanCallback      OIDCCallback
+	AwsCredentialsProvider func(context.Context) (aws.Credentials, error)
```

Comment:
Initialisms in Go are conventionally all caps. This should be `AWSCredentialsProvider`.